File: app/python/lib/stage.py
"""ステージ管理クラス"""

from lib.config import Config
from lib.puyoimage import PuyoImage


class Stage:
    """ゲームステージ（盤面）を管理するクラス"""

    def __init__(self, config: Config, puyo_image: PuyoImage) -> None:
        """ステージの初期化

        Args:
            config: ゲーム設定
            puyo_image: ぷよ画像
        """
        self.config = config
        self.puyo_image = puyo_image

File: app/fsharp-2/fable/src/Stage.fs
// src/Stage.fs
module PuyoPuyo.Stage

open PuyoPuyo.Types

/// ステージの状態
type StageState = {
    Cells: PuyoColor array array
    Rows: int
    Cols: int
}

/// ステージを作成する
let create () : StageState =
    let rows = 12
    let cols = 6
    let cells = Array.init rows (fun _ -> Array.create cols PuyoColor.Empty)
    {
        Cells = cells
        Rows = rows
        Cols = cols
    }
